File: frontend/src/types/rental.ts
import { Copy } from "./copy";
import { User } from "./user";

export interface Rental {
    user: User,
    copy: Copy,
    startDate: Date,
    dueDate: Date,
    returned: boolean
}

export type NewRental = { 
    userId: number,
    copyId: number
}

export type UpdateRental = {
    id: number,
    userId: number,
    returned: boolean,
    dueDate: Date
}

File: backend/Models/Rental.cs
using System.ComponentModel.DataAnnotations;
using Library.DTOs;

namespace Library.Models;

public class Rental : BaseModel
{
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    public UserRegisterResponseDTO? UserInfo => User != null ? UserRegisterResponseDTO.FromUser(User) : null;
    public int CopyId { get; set; }
    public Copy Copy { get; set; } = null!;
    [Required]
    public DateTime DateRented { get; set; }
    public bool Returned { get; set; } = false;
    [Required]
    public DateTime DueDate {get; set;}
}